pkg/gameserverallocations/controller.go

```diff
@@ -386,18 +495,143 @@ func (c *Controller) findReadyGameServerForAllocation(gsa *v1alpha1.GameServerAl
 		}
 	}
 
+	bestGSList := []v1alpha1.GameServer{}
```

Comment:
My only concern here, is that we get more of a random pattern of gameservers - unless I'm reading this wrong (which I may be), we aren't grabbing an ordered list of GameServers based on comparators, we are grabbing N number of the gameserver that we come across (in fact, the comparator isn't really doing anything at this point).

So we're breaking the `Scheduling` contract here essentially. While I see we want to break it a little (pull a random one out of set), I feel like we should at least get the top N number of sorted gameservers by comparator.

The solution I think would be easiest (although probably not the fastest) - would be to sort the `allocationSet` based on the `nodeCount` values that the allocation comes from with the `comparator`, and then take the top N GameServers from that. WDYT? Or  is there some way we could pre-calculate this? (probably not with the combo of selectors).

Reply:
I don't _think_ this gives us a sorted list, but it gives us a slightly better list than before (since list ordering out of the informer cache is random, and we aren't going through all gameservers before appending them).

The `Packed` scheduling is in place, because without it, you get the "swiss cheese" problem, wherein you can't shut down nodes because of sparse game servers -- which at scale is real money.

> I can check-in by removing else if & else (see above) so we don't break the contract and we can create another issue to improve this. WDYT?

I think that makes the most sense. I've also realise that the recent changes to gameserverset also break the contract -- so I'm wondering :thinking: if it makes sense to keep an ordered by node count list of gameservers in a cache somewhere, that can be used globally, and are updated at event time -- but agreed, that should be a separate PR.

So that sounds like a good way through :+1: to get these changes over the line.

Reply:
Ok, I will remove the else if & else to not break the Scheduling contract.
Just FYI, the performance difference between breaking and not breaking the contract is 175 Allocations per sec vs 116 Allocations per sec (first load test is which breaks the contract).
![image](https://user-images.githubusercontent.com/17522001/53395777-894d6700-3957-11e9-854b-7eedb82d5388.png)